### PR TITLE
GooseFX is now compatible with the Solana Mobile Wallet Adapter protocol

### PR DESCRIPTION
**Pull request by steveluscher**
# Description

In this PR we filter out MWA if it's determined not to be supported, and make it so that, when the MWA is selected, clicking the connect button starts a connection instead of popping open the dialog.

### Mobile, with mobile wallet

https://user-images.githubusercontent.com/13243/199039264-5d1ffdee-1f1b-4f28-a98b-d838abde55a9.mov

### Desktop

https://user-images.githubusercontent.com/13243/198397288-bca1632d-0d9c-4be0-8fae-c2a9b2a747d8.mov

### CODE REVIEW

File: src/utils/wallets.ts [hunk 1]
```diff
@@ -11,11 +11,20 @@ import {
...
 } from '@solana-mobile/wallet-adapter-mobile'
 
 export const getWalletAdapters = (network: WalletAdapterNetwork) => [
+  new SolanaMobileWalletAdapter({
+    appIdentity: { name: 'Goosefx App' },
+    authorizationResultCache: createDefaultAuthorizationResultCache(),
+    addressSelector: createDefaultAddressSelector(),
+    cluster: network,
+    onWalletNotFound: createDefaultWalletNotFoundHandler()
+  }),
```
Comment (steveluscher) [inline src/utils/wallets.ts]: I moved this to the top of the list _because_ it gets filtered out:

* on desktop
* when on a non-supported mobile device (eg. iOS)

You can see this in the desktop test plan video.

**Comment by walt-1** [comment]
Hi @steveluscher - I believe these changes have already been pulled into a working branch via [PR 532](https://github.com/GooseFX1/gfx-web-app/pull/532)?

There are some changes I need to make regarding custom connection logic we have on our wallets. That is why I currently have those changes in a working branch, not in dev at moment.

If I need to disregard that PR and merge this, let me know. I will need to change the base branch to make those changes.

**Comment by steveluscher** [comment]
> I believe these changes have already been pulled into a working branch via https://github.com/GooseFX1/gfx-web-app/pull/532?

Not quite! These are Mobile Wallet Adapter specific fixes that follow from that PR.

**Comment by walt-1** [comment]
Since your changes are based on a branch that do not reflect Jordan's implementations, it is hard to decipher what changes are needed to promote. Specifically regarding the dependencies. 

Jordan has changed react/types versions, solana package versions, and "@solana/wallet-adapter-base", "@solana/wallet-adapter-react", and "@solana/wallet-adapter-wallets" have more up to date versions.

Can you take a look at the package.json file with his branch being the base to sort out the dependencies/versions?

**Comment by steveluscher** [comment]
Sure thing! I just mashed ‘fork’ presuming that Jordan’s changes were already merged. I’ll rebase a bit later!

**Comment by steveluscher** [comment]
Done, @lukewalt! I updated this PR, with `mob-wallet-adapter-wip` as the base.

**Comment by walt-1** [comment]
Thanks @steveluscher - I will keep an eye on the status of this PR and will merge when ready!

**Comment by steveluscher** [comment]
Alright! I retested this and believe it's good to go. Review welcome!

**Review by walt-1 — APPROVED** [review]

**Pull request merged**